src/bootstrap/bootstrap.py

```diff
@@ -1068,6 +1123,12 @@ def bootstrap(help_triggered):
 
     # Fetch/build the bootstrap
     build.build = args.build or build.build_triple()
+    build.stage1_commit = build.maybe_download_stage1()
+    if build.stage1_commit is not None:
+        if build.verbose:
+            commit = build.stage1_commit
+            print("using downloaded stage1 artifacts from CI (commit {})".format(commit))
+        build.rustc_channel = "nightly"
```

Comment:
This seems suspicious - the stage0.txt channel should not be nightly, it shouldn't be touched at all?

Reply:
This is needed for https://github.com/rust-lang/rust/blob/d0aa330f67080f07b77b2c58629b85572626dcf8/src/bootstrap/bootstrap.py#L402-L403, so it downloads `rust-std-nightly-$TARGET.tar.xz` instead of `-beta-` or something like that.

Reply:
Ah, I see. So it's not actually true that this must be nightly. e.g., on beta branch it will be beta. Can we leave a FIXME here? I'm not too worried about actually supporting beta in this PR, since that is hard, but leaving a comment seems appropriate.